### PR TITLE
feat: Optimised unnecessary Histogram copy out.

### PR DESCRIPTION
**Pull request by ItsLifeJim**
Optimised unnecessary Histogram copy out.

SDB-5759

### CODE REVIEW

File: toolbox/io/Runner.cpp [hunk 1]
```diff
@@ -48,14 +49,16 @@ void run_reactor(Reactor& r, long busy_cycles, ThreadConfig config, const std::a
 }
 
 static const std::chrono::seconds metric_interval = 60s;
+Histogram* make_histogram() { return new Histogram{1, 1000, 2}; } // 100% accurate to 256
```
Comment (markaylett) [inline toolbox/io/Runner.cpp:52]: This should return a `unique_ptr<>`.

File: toolbox/io/Runner.cpp [hunk 2]
```diff
@@ -48,14 +52,22 @@ void run_reactor(Reactor& r, long busy_cycles, ThreadConfig config, const std::a
...
+                                  const std::string& metric_id)
+{
+    // Histogram is 100% accurate to 256, covering Reactor::MaxEvents of 128 work items.
+    return HistogramMetricPtr{new HistogramMetric{app_name, metric_type, metric_id, 1, 1000, 2}};
+}
+
 void run_metrics_reactor(Reactor& r, long busy_cycles, ThreadConfig config,
-                         const std::atomic<bool>& stop, MetricCallbackFunction metric_callback)
+                         const std::atomic<bool>& stop, MetricCallbackFunction metric_callback,
+                         const std::string app_name, const std::string metric_type, const std::string metric_id)
```
Comment (markaylett) [inline toolbox/io/Runner.cpp:64]: String params should either be const-ref or non-const+move-from.
@Shivix can you help Bill with some of the modern C++ patterns/idioms?

Bear in mind that toolbox is intended to be a suite of general purpose, open-source libraries for developing applications in our industry/domain.
It ideally should not assume any Reactive specifics/concepts such as app_name, metric_type, etc.
Anything specific to Switchboard or organisation-specific should be moved to reactive-cpp.

Comment (ItsLifeJim) [reply toolbox/io/Runner.cpp:64]: Thanks Mark.  That's pretty much how I had the previous version, with just the Histogram cycling in the Reactor Runner and the Reactive code decorating the metrics, but changed it to follow your suggestion. I'll work with Mo to find a balance whereby the Reactor callback isn't too expensive and we have a clear separation of concerns.

Comment (markaylett) [reply toolbox/io/Runner.cpp:64]: The change is mostly good.
The only two aspects I am questioning are:

1. `const std::string` as the parameter type;
2. adding `HistogramMetric` and extending ReactorRunner with Switchboard specifics.

One solution to #2 is for reactive-cpp to simply have its own version of ReactorRunner. We may also conclude that the strict separation I suggest is overkill and we should just leave as is for now.

Comment (ItsLifeJim) [reply toolbox/io/Runner.cpp]: Swapped to string_view throughout

**Pull request closed without merging**